macros/src/main/scala/clue/macros/GraphQL.scala

```diff
@@ -0,0 +1,356 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package clue.macros
+
+import cats.syntax.all._
+import scala.annotation.StaticAnnotation
+import scala.reflect.macros.blackbox
+import scala.annotation.tailrec
+import edu.gemini.grackle._
+import edu.gemini.grackle.{ Type => GType }
+import edu.gemini.grackle.{ NoType => GNoType }
+import edu.gemini.grackle.{ TypeRef => GTypeRef }
+import edu.gemini.grackle.UntypedOperation._
+import cats.effect.IO
+
+class GraphQL(
+  val mappings: Map[String, String] = Map.empty,
+  val eq:       Boolean = false,
+  val show:     Boolean = false,
+  val lenses:   Boolean = false,
+  val reuse:    Boolean = false,
+  val debug:    Boolean = false
+) extends StaticAnnotation {
+  def macroTransform(annottees: Any*): Any = macro GraphQLImpl.resolve
+}
+private[clue] final class GraphQLImpl(val c: blackbox.Context) extends GraphQLMacro {
+  import c.universe._
+
+  final def resolve(annottees: Tree*): Tree =
+    macroResolve(annottees: _*)
+
+  /**
+   * Extract the `document` contents from the `GraphQLQuery` the marcro was applied to.
+   */
+  @tailrec
+  private[this] def documentDef(tree: List[Tree]): Option[String] =
+    tree match {
+      case Nil                                          => none
+      case q"$mods val document: $tpe = $document" :: _ =>
```

Comment:
is the content of a quasiquote compiled?
Can you put illegal code inside it?

Reply:
Quasiquotes translate into ASTs. They do have to express a valid AST or the macro won't compile. That said, nothing prevents you from building an illegal AST.